pglogical/hooks.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022 Hunki Enterprises BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+from odoo.sql_db import Cursor
+from odoo.tools import config
+
+SECTION_NAME = "pglogical"
+
+
+def post_load():
+    """
+    Patch cursor to funnel DDL through pglogical.replicate_ddl_command if configured to
+    do so
+    """
+
+    _logger = logging.getLogger("odoo.addons.pglogical")
+
+    if SECTION_NAME not in config.misc:
+        _logger.info("%s section missing in config, not doing anything", SECTION_NAME)
+        return
+    replication_sets = list(
+        filter(None, config.misc[SECTION_NAME].get("replication_sets", "").split(","))
+    )
+    if not replication_sets:
+        _logger.error("no replication sets defined, not doing anything")
+        return
+    if config["test_enable"]:
+        _logger.info("test mode enabled, not doing anything")
+        return
+
+    _logger.info("patching cursor to intercept ddl")
+    execute_org = Cursor.execute
+
+    def execute(self, query, params=None, log_exceptions=None):
+        """Wrap DDL in pglogical.replicate_ddl_command"""
+        if (query[:6] == "CREATE" or query[:5] == "ALTER" or query[:4] == "DROP") and (
```

Comment:
Super nice work @hbrunn!

For recognizing DDL we could also use `sqlparse` to iterate through multiple statements if you think it's needed, [get_type()]. (though likely not necessary, I assume it should be rare to have multiple statements in)

Reply:
I think we could use that in the second stage for detecting the commands we want to ignore. But the first conditional should be fast, this function is called *often*

Reply:
Maybe Odoo is consistent in its shouting style of SQL formatting, but the casing in migration scripts might vary. What do you think of applying `upper` to at least those first characters of the query that you are checking?

Reply:
I added a somewhat different heuristics now, look at it more closely if it doesn't start with 'SELECT' - this way we don't impact select statements at all, and everything else gets the accurate treatment with the `sqlparse` lib.

Reply:
Would it be safe to exclude queries starting with UPDATE as well?

Reply:
That's a consideration of course. I'm not sure if it's going to be noticable, but it's a relatively large class of queries that we can save some overhead on.

Reply:
I've added UPDATE